backend/tests/test_bridge_host.py

```
@@ -631,3 +631,625 @@ def dyn(a: int) -> dict:
+# nested case would pass without exercising the recursion at all.
+_NESTED_INCOMPATIBILITY = {"dependencies": {"a": ["b"]}}
+for _kw in host._SCHEMA_KEYWORDS:
+    if _kw == "items" or _kw in host._POST_DRAFT_07_ASSERTION_KEYWORDS:
+        continue
+    _INCOMPATIBLE_CONSTRUCT_CASES.append(
+        ({_kw: _NESTED_INCOMPATIBILITY}, True, f"nested-under-{_kw}")
+    )
+for _kw in host._SCHEMA_LIST_KEYWORDS:
+    _INCOMPATIBLE_CONSTRUCT_CASES.append(
+        ({_kw: [_NESTED_INCOMPATIBILITY]}, True, f"nested-under-{_kw}")
+    )
```

Ruff (0.16.1)

[warning] Use `list.extend` to create a transformed list

(PERF401)